### PR TITLE
Add c-progressbar

### PR DESCRIPTION
**Pull request by Horta**

### CODE REVIEW

**Comment by Horta** [comment]
ready for approval =)

File: recipes/c-progressbar/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  url: https://github.com/limix/progressbar/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - python  # [win]
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: If we are skipping Windows, this can be dropped.

File: recipes/c-progressbar/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  requires:
+    - python  # [win]
+  commands:
+    - test -e $PREFIX/lib/libprogressbar.so  # [linux]
+    - test -e $PREFIX/lib/libprogressbar_static.a  # [linux]
+    - test -e $PREFIX/lib/libprogressbar.dylib  # [osx]
+    - test -e $PREFIX/lib/libprogressbar_static.a  # [osx]
+    - if NOT exist %PREFIX%/lib/libprogressbar.lib exit 1  # [win]
+    - if NOT exist %PREFIX%/lib/libprogressbar.dll exit 1  # [win]
+    - if NOT exist %PREFIX%/lib/libprogressbar_static.lib exit 1  # [win]
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: No need for the Windows tests if we aren't building for it.

File: recipes/c-progressbar/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - conda-build
+    - toolchain
+    - cmake
+    - ncurses
+  run:
+    - ncurses
+
+test:
+  requires:
+    - python  # [win]
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: This can also be dropped.

File: recipes/c-progressbar/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+requirements:
+  build:
+    - python  # [win]
+    - conda-build
+    - toolchain
+    - cmake
+    - ncurses
+  run:
+    - ncurses
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: Please pin build and run time dependencies according to the [pinning script]( https://github.com/conda-forge/conda-forge.github.io/blob/d2cd7bc2c8bbfc83e7cbc11a32379fb9b77c0883/scripts/pin_the_slow_way.py#L40-L81 ).

File: recipes/c-progressbar/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - python  # [win]
+    - conda-build
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: Why is `conda-build` a build requirement?

File: recipes/c-progressbar/bld.bat [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+         -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE ^
+         -DCMAKE_INSTALL_PREFIX:PATH=%PREFIX% ^
+         ..
+if errorlevel 1 exit 1
+nmake
+if errorlevel 1 exit 1
+nmake test
+if errorlevel 1 exit 1
+nmake install
+if errorlevel 1 exit 1
```
Comment (jakirkham) [inline recipes/c-progressbar/bld.bat]: Please drop this file if we are not building on Windows. Can always readd all of this with a PR to the feedstock.

File: recipes/c-progressbar/build.sh [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+export CFLAGS="${CFLAGS} -O3 -fPIC"
+export LDFLAGS="${LDFLAGS} -Wl,-rpath,${PREFIX}/lib"
+
+mkdir build
+cd build
+
+cmake -DCMAKE_INSTALL_PREFIX:PATH=$PREFIX ..
+make -j$CPU_COUNT
+make install
```
Comment (jakirkham) [inline recipes/c-progressbar/build.sh]: Is there a `make check` that can be run before installing?

File: recipes/c-progressbar/meta.yaml (hunk 2 above)
Comment (Horta) [inline recipes/c-progressbar/meta.yaml]: sorry, fixed! I actually want windows...

File: recipes/c-progressbar/build.sh (hunk 7 above)
Comment (Horta) [inline recipes/c-progressbar/build.sh]: done!

File: recipes/c-progressbar/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/limix/progressbar/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: This shouldn't be a bare skip, right?

File: recipes/c-progressbar/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: Well I don't think we have `ncurses` for Windows. 😕

File: recipes/c-progressbar/meta.yaml (hunk 8 above)
Comment (Horta) [inline recipes/c-progressbar/meta.yaml]: ops

**Comment by jakirkham** [comment]
AppVeyor is struggling with rate limiting. That said, I don't think this can build on Windows because we don't have `ncurses` on Windows. Unless, of course, `ncurses` is a soft requirement and can be dropped on Windows.

**Comment by Horta** [comment]
Bump!

**Comment by mariusvniekerk** [comment]
@jakirkham ping

**Comment by jakirkham** [comment]
Sorry I don't have time to look at this. Please ping staged-recipes to complete the review.

**Comment by mariusvniekerk** [comment]
ping @conda-forge/staged-recipes

File: recipes/c-progressbar/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/limix/progressbar/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win and py27]
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: For Windows, this needs `vc` features.

```yaml
  features:
    - vc9   # [win and py27]
    - vc10  # [win and py34]
    - vc14  # [win and py35]
    - vc14  # [win and py36]
```

File: recipes/c-progressbar/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    - python  # [win]
+    - toolchain
+    - cmake
+    - ncurses 5.9  # [not win]
+  run:
+    - ncurses 5.9  # [not win]
+
+test:
+  requires:
+    - python  # [win]
```
Comment (jakirkham) [inline recipes/c-progressbar/meta.yaml]: Should be.

```yaml
    - python {{ environ['PY_VER'] + '*' }}  # [win]
```

**Pull request merged**